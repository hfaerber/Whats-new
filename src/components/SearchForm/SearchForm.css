header {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  color: whitesmoke;
  grid-area: header;
  padding: 0 19px;
  margin-left: 25px;
}

form {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  width:82%;
  margin-left: -20px;
}

.inputLabel {
  width: 50%;
  /* border: 1px solid yellow; */
  text-align: left;
  font-size: 1.2em;
}

.formInput {
  padding: 10px;
  font-size: 1em;

  width: 100%;
  background-color: #141527;
  color: whitesmoke;
  border: none;
  display: block;
  border-radius: 5px;
  /* border: 1px solid yellow; */
  box-shadow: inset 0 0 10px #fff;
}

.formButton {
  padding: 0 10px;
  height: 42px;
  width: 20%;
  cursor: pointer;
  border-radius: 20px;
  font-size: 1.2em;
}
