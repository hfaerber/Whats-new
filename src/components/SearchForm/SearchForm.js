import React, {Component} from 'react';
import './SearchForm.css';
import PropTypes from 'prop-types';

class SearchForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      searchQuery: ''
    }
  }

  handleChange = event => {
    this.setState({ searchQuery: event.target.value })
  }

  handleSearchButton = event => {
    this.props.filterBySearch(event, this.state.searchQuery);
    this.resetSearchField();
  }

  resetSearchField = () => {
    this.setState({
      searchQuery: ''
    })
  }

  render() {
    return(
      <header>
        <h1>What's new?</h1>
        <form>
        <label className='inputLabel'>Search for news articles here
        <input className='formInput' type='text'
          placeholder=' '
          value={this.state.searchQuery}
          onChange={event => this.handleChange(event)}
          />
        </label>
        <button className="formButton"
          onClick={event => this.handleSearchButton(event)}>Search Now</button>
        </form>
      </header>
    )
  }
}

SearchForm.propTypes = {
  filterBySearch: PropTypes.func,
}

export default SearchForm;
