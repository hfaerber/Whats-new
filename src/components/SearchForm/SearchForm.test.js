import React from 'react';
import SearchForm from './SearchForm.js';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<SearchForm
      filterBySearch={jest.fn()}
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockFilterBySearch = jest.fn();
    const mockEvent = { target: { value: 'pants' } };
    const wrapper = shallow(<SearchForm
      filterBySearch={mockFilterBySearch}
      />);
    const expected = 'pants';

    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('searchQuery')).toEqual(expected);
  })

  it('should invoke handleChange when changes occur', () => {
    const mockEvent = { target: { value: 'pants' }}
    const wrapper = shallow(<SearchForm
      filterBySearch={jest.fn()}
    />);
    wrapper.instance().handleChange = jest.fn();

    wrapper.find('input').simulate('change', mockEvent)
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent);
  })

  it('should call the filterBySearch prop with searchQuery state when clicked',
    () => {
      const mockFilterBySearch = jest.fn();
      const mockEvent = {};
      const wrapper = shallow(<SearchForm
          filterBySearch={mockFilterBySearch}
        />);
      const mockSearchQuery = 'test';

      wrapper.setState({searchQuery: mockSearchQuery});
      wrapper.find('button').simulate('click', mockEvent)
      expect(mockFilterBySearch).toHaveBeenCalledWith(mockEvent, mockSearchQuery)
  })

})
