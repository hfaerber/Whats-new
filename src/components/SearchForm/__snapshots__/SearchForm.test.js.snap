// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot with all data passed in correctly 1`] = `
<header>
  <h1>
    What's new?
  </h1>
  <form>
    <label
      className="inputLabel"
    >
      Search for news articles here
      <input
        className="formInput"
        onChange={[Function]}
        placeholder=" "
        type="text"
        value=""
      />
    </label>
    <button
      className="formButton"
      onClick={[Function]}
    >
      Search Now
    </button>
  </form>
</header>
`;
